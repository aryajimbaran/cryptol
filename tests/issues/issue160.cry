thm1 : [16] -> Bit
property thm1 x = (x <<< 16 == x) && (x >>> 16 == x)

thm2 : [16] -> [16] -> Bit
property thm2 x i = x >>> i == x >>> (i && 0x000f)

thm3 : [15] -> [8] -> Bit
property thm3 x i = x <<< i == x <<< (i % 15)

thm4 : [15] -> [15] -> Bit
property thm4 x i = x <<< i >>> i == x

thm5 : [15] -> [3] -> Bit
property thm5 x i = x <<< i >>> i == x

thm6 : [8] -> [4] -> [4] -> Bit
property thm6 x i j = x <<< i >>> j == x <<< (i - j)
