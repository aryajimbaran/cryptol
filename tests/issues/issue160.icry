:l issue160.cry
:prove
