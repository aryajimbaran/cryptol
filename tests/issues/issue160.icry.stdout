Loading module Cryptol
Loading module Cryptol
Loading module Main
property thm1 Q.E.D.
property thm2 Q.E.D.
property thm3 Q.E.D.
property thm4 Q.E.D.
property thm5 Q.E.D.
property thm6 Q.E.D.
